src/Bank/BankLogic/BankAccount.cs

```diff
@@ -4,16 +4,16 @@ namespace BankingSystem.BankLogic
 {
     public abstract class BankAccount : IBankAccount
     {
-        protected BankAccount(int accountNumber, AccountOrigin accountOrigin, double interestRate)
+        protected BankAccount(int accountNumber, AccountOrigin origin, double interestRate)
         {
             Balance = 0;
             AccountNumber = accountNumber;
-            AccountOrigin = accountOrigin;
+            OriginOwner = Owner.GetOriginOwner(origin);
```

Comment:
this should be at least `origin.GetOwner() `

Reply:
also rename the variable to be owner

Reply:
;)